tests/generate-testdata.js

```diff
@@ -0,0 +1,250 @@
+      // '@context:integer': 4,
+      // '@context:null': null,
+      '@context:object': { '@vocab': 'https://www.w3.org/2018/credentials/v1/#' },
+      '@context:string': 'https://www.w3.org/2018/credentials/v1',
+    };
+    for (const [description, invalidValue] of Object.entries(invalidValues)) {
+      const document = klona(baseCredential);
+      document['@context'] = invalidValue;
+      sampleVCs[description] = await addProof({ document });
+    }
+  }
+
+  {
+    const invalidValues = {
+      // '@context:item:array': ['https://www.w3.org/2018/credentials/v1', []],
+      // '@context:item:boolean': ['https://www.w3.org/2018/credentials/v1', false],
+      // '@context:item:integer': ['https://www.w3.org/2018/credentials/v1', 4],
+      // '@context:item:null': ['https://www.w3.org/2018/credentials/v1', null],
+      '@context:item:object': [
+        'https://www.w3.org/2018/credentials/v1',
+        { '@vocab': 'https://www.w3.org/2018/credentials/v1/#' },
+      ],
+    };
+    for (const [description, invalidValue] of Object.entries(invalidValues)) {
+      const document = klona(baseCredential);
+      document['@context'] = invalidValue;
+      sampleVCs[description] = await addProof({ document });
+    }
+  }
+
+  {
+    const invalidValues = {
+      // 'id:array': ['urn:uuid:07aa969e-b40d-4c1b-ab46-ded252003ded'],
+      // 'id:boolean': false,
+      // 'id:integer': 123,
+      // 'id:null': null,
+      // 'id:object': { key: 'urn:uuid:07aa969e-b40d-4c1b-ab46-ded252003ded' },
+    };
+    for (const [description, invalidValue] of Object.entries(invalidValues)) {
+      const document = klona(baseCredential);
+      document.id = invalidValue;
+      sampleVCs[description] = await addProof({ document });
+    }
+  }
+
+  {
+    const description = 'type:missing';
+    const document = klona(baseCredential);
+    delete document.type;
+    sampleVCs[description] = await addProof({ document });
+  }
+
+  {
+    const invalidValues = {
+      // 'type:boolean': false,
+      // 'type:integer': 123,
+      // 'type:null': null,
+      // 'type:object': { key: 'VerifiableCredential' },
+      'type:string': 'VerifiableCredential',
+      // 'type:item:array': ['VerifiableCredential', []],
+      // 'type:item:boolean': ['VerifiableCredential', false],
+      // 'type:item:integer': ['VerifiableCredential', 4],
+      // 'type:item:null': ['VerifiableCredential', null],
+      // 'type:item:object': ['VerifiableCredential', { foo: true }],
+    };
+    for (const [description, invalidValue] of Object.entries(invalidValues)) {
+      const document = klona(baseCredential);
+      document.type = invalidValue;
+      sampleVCs[description] = await addProof({ document });
+    }
+  }
+
+  {
+    const description = 'issuer:missing';
+    const document = klona(baseCredential);
+    delete document.issuer;
+    sampleVCs[description] = await addProof({ document });
+  }
+
+  {
+    const invalidValues = {
```

Comment:
BTW this looks like a good case for a map:

```js
const invaludValues = new Map([
  ['issuer:array', ['did:example:123']],
 ['issuer:boolean', false]
])
```

then you get an iterable by default

Also you might want to change the VC context here to allow for the invalid values

```js
  "issuer": {"@id": "cred:issuer", "@type": "@id"},
```

```js
        "issuer": {"@id": "cred:issuer", "@type": "xsd:Boolean"}
```

Reply:
@aljones15 

When I modify `issuer` in the context in this way, and try to set `issuer: {"id": false}` in the credential, I still get the same error I get without modifying the context:
```
JsonLdError [jsonld.SyntaxError]: Invalid JSON-LD syntax; "@id" value must a string.
```
Note that setting `issuer: false` does not cause any errors, nor does `issuer: {"id": "did:web:example.com"}`

Reply:
This appears to be working:
```
        context['@context'].VerifiableCredential['@context'].issuer = {
          '@id': 'cred:issuer',
          '@type': '@id',
          '@context': {
            id: 'xsd:boolean',
          },
        };
```

Reply:
Mutating any `type` property or the `@context` to anything invalid continues to prove difficult.

Reply:
> Mutating any type property or the @context to anything invalid continues to prove difficult.

I just have to say ... that's great! :P ... the above is a sign of a good implementation / reasonably safe APIs. We want it to be hard to create invalid stuff -- since the only use case is a negative test suite.

Reply:
@dlongley -- Your turn to have an un-fenced `@context` in the quoted content that starts https://github.com/w3c-ccg/traceability-interop/pull/400#discussion_r975702456. Please edit to add code fences.